### ISSUE TITLE
SGgame - Reimplement Type Script

### ISSUE THREAD

**Issue by HoeenCoder** [HoeenCoder/Wavelength#67]
Typescript was disabled for the SGgame branch temporarly, until i can learn it and implement things for wavelength. This is mostly a note so I don't forget to re-add it.

```json
"scripts": {
    "start": "node pokemon-showdown",
    "test": "eslint --rulesdir=dev-tools/eslint --cache . && tsc && mocha"
  },
```

**Comment by DeathlyPlays** [comment]
I suck at TypeScript so far I don't know what's up with its eslint

**Comment by PowerHitter0418** [comment]
I'm not that good at TypeScript but this might work as far as I know?  
``{
    "start";
    "node pokemon-showdown",
        "test";
    "eslint --rulesdir=dev-tools/eslint --cache . && tsc && mocha";
}``

**Comment by HoeenCoder** [comment]
@PowerHitter0418 yes that re-enables it in the sense that it tests typescript, but the tests will fail. We also need to typescript the SGgame branch.

**Comment by Volco** [comment]
If we REALLY must learn typescript @HoeenCoder i can try

**Comment by AHelpfulRayquaza** [comment]
Ohk (PowerHitter here btw)

**Comment by HoeenCoder** [comment]
This is low priority, so don't stress over it. Might just cancel it anyways.

**Comment by Volco** [comment]
Mk
